fact/src/output.rs

```diff
@@ -0,0 +1,87 @@
+use std::{path::PathBuf, sync::Arc};
+
+use log::{info, warn};
+use tokio::{
+    sync::{broadcast, watch},
+    task::JoinHandle,
+};
+
+use crate::{config::FactConfig, event::Event, grpc};
+
+pub struct Output {
```

Comment:
There's possibly an argument that this component should be split into two parts - (1) the bit that handles the tokio spawning and reading from the Receiver and (2) the bit that handles the implementation-specific details about where the events should be sent (grpc or stdout currently.)

There's a bit of duplication between the `start_*` methods in this existing implementation which we could avoid by splitting this up, but it's not a significant enough problem to bother with refactoring it now IMO